inc/c_util/tarray_ll.h

```diff
@@ -99,6 +105,51 @@ TARRAY_LL(T) TARRAY_LL_CREATE(C)(void)
     return result;                                                                                                          \
 }
 
+/*introduces a function definition for tarray_create_with_capacity*/
+#define TARRAY_LL_CREATE_WITH_CAPACITY_DEFINE(C, T)                                                                         \
+TARRAY_LL(T) TARRAY_LL_CREATE_WITH_CAPACITY(C)(uint32_t capacity)                                                           \
+{                                                                                                                           \
+    TARRAY_TYPEDEF_NAME(T)* result;                                                                                         \
+    if (capacity == 0)                                                                                                      \
+    {                                                                                                                       \
+        /* Codes_SRS_TARRAY_01_001: [ If capacity is 0, TARRAY_CREATE_WITH_CAPACITY(T) shall fail and return NULL. ]*/      \
+        LogError("Invalid argumenets: uint32_t capacity=%" PRIu32 "", capacity);                                            \
```

Comment:
>argumenets [](http://example.com/codeflow?start=26&length=10)

I suppose these are some sort of juicy HARIBOes.

Reply:
No more HARIBOs!